service/order.go

```diff
@@ -273,8 +293,13 @@ func (s *OrderService) Delete(ctx context.Context, in *pb.OrderSpecificRequest)
 		return nil, errors.E(errors.Op("Delete order"), err)
 	}
 
+	sig, err := identity.Sign(s.Storage, orderInBytes)
```

Comment:
This shouldn't push the delete wireMessage to the network if you're not the creator. NOTE: You can remove any order you want, but not propagate it to the network if you're not the owner.